src/components/_data/ContentSingularData.ts

```diff
@@ -1,6 +1,5 @@
 export type ContentSingularData = {
     plainText_content?: string
-    exampleImage_imageLink?: string
-    exampleImage_percentageSize?: number
+    toDoList?: {index?: number, task?: string}[]
```

Comment:
I would declare a type for a todo item first, then do something like `ToDoItem[]`, that way if you ever need the types for an individual `ToDoItem` you can get the type and enforce that it has a key index and task. 

I would also make them not have `?`s, because every todo item needs one of those, and it should never be undefined otherwise things mess up. You'll have extra unnecessary logic checking for undefined.